examples/fNIRS/classify_fNIRS.py

```diff
@@ -0,0 +1,348 @@
+        else:
+            raise ValueError("Parameter shrinkage must be None, a float, or a list of floats.")
+
+        # Lists of supported metrics
+        kernel_metrics = [
+            'linear', 'poly', 'polynomial', 'rbf',
+            'laplacian', 'sigmoid', 'cosine'
+        ]
+        covariance_estimators = [
+            'scm', 'lwf', 'oas', 'mcd', 'corr', 'tyl', 'np', 'riemann',
+            'identity', 'cov'
+        ]
+
+        M_matrices = []
+
+        for i in range(n_samples):
+            start = 0
+            M_blocks = []
+            for idx, (block_size, metric, shrinkage_value) in enumerate(
+                    zip(blocks, metrics, shrinkages)):
+                end = start + block_size
+                # Extract the block of channels
+                X_block = X[i, start:end, :]  # shape: (block_size, n_times)
+
+                # Compute the matrix for this block
+                if metric in kernel_metrics:
+                    # Compute kernel matrix
+                    M_block = pairwise_kernels(
+                        X_block,
+                        metric=metric,
+                        n_jobs=self.n_jobs,
+                        **self.kwds
+                    )
+                elif metric in covariance_estimators:
+                    # Compute covariance matrix
+                    X_block_reshaped = X_block[np.newaxis, :, :]
+                    M_block = covariances(
+                        X_block_reshaped, estimator=metric, **self.kwds
+                    )[0]
+                else:
+                    raise ValueError(
+                        f"Metric '{metric}' is not recognized as a kernel metric "
+                        f"or covariance estimator."
+                    )
+
+                # Apply shrinkage if specified
+                if shrinkage_value is not None:
+                    M_block_reshaped = M_block[np.newaxis, :, :]
+                    shr = Shrinkage(shrinkage=shrinkage_value)
+                    M_block = shr.fit_transform(M_block_reshaped)[0]
+
+                M_blocks.append(M_block)
+                start = end
+
+            # Create the block diagonal matrix
+            M_full = self._block_diag(M_blocks)
+            M_matrices.append(M_full)
+
+        return np.array(M_matrices)
+
+    @staticmethod
+    def _block_diag(matrices):
+        """Construct a block diagonal matrix from a list of square matrices."""
+        shapes = [m.shape[0] for m in matrices]
+        total_size = sum(shapes)
+        result = np.zeros((total_size, total_size), dtype=matrices[0].dtype)
+        start = 0
+        for m in matrices:
+            end = start + m.shape[0]
+            result[start:end, start:end] = m
+            start = end
+        return result
+
+###############################################################################
+# Load data
+# ---------
+
+print("Loading preprocessed fNIRS data")
+# Adjust the path to your data
+data_path = './data'
```

Comment:
I doubt this will work as our doc is rendered automatically and data should not be commited. You need the data to be fetched from some website on run.